packages/discovery-provider/src/api/v1/tracks.py

```diff
@@ -757,7 +609,10 @@ def get(self, track_id):
                 cached_content_node, path, skip_check
             )
             if stream_url:
-                return stream_url
+                if no_redirect:
+                    return success_response(stream_url)
+                else:
+                    return stream_url
```

Comment:
If I understand the behavior of the `cache` decorator with the `transform=redirect` argument:
* If this function returns a tuple (which `success_response` will create), the decorator does not run the transform function
* If it returns a single value, the transform is run on said value
* In both cases, the cache decorator will cache the value _iff_ there is no `user_id` present in the original request. And it will only attempt to use the cached values if the `user_id` param isn't present on the request.

Assuming that's correct, it might be a good idea to test this endpoint both with/without `no_redirect` and with/without passing a `user_id` to make sure it does what we expect in all cases.


And also the hidden magic behavior of the cache decorator smells a little bit 😅

Reply:
I'm very far from an expert, but the first couple bullets is my limited understanding as well. 
It does work as expected in all 4 cases you mentioned (with/without user_id & no_redirect). 
I'm not sure about the `user_id` & cache bit that you mentioned.

Reply:
Not as worried about the user_id and cache bit. I think the expected behavior there would be that if you hit the endpoint while logged out, you would expect a subsequent request to the same endpoint to be faster due to hitting the cache.